### PR TITLE
Refactor QueryParams

### PR DESCRIPTION
**Pull request by Feiyang1**
I kept simple getters and `copy()` within the class. Other methods are changed to functions.

### CODE REVIEW

File: packages/database/src/core/view/QueryParams.ts [hunk 1]
```diff
@@ -178,7 +174,18 @@ export class QueryParams {
     return this.index_;
   }
 
-  private copy_(): QueryParams {
+  loadsAllData(): boolean {
+    return !(this.startSet_ || this.endSet_ || this.limitSet_);
+  }
+
+  /**
+   * What does isDefault() mean?
```
Comment (hsubox76) [inline packages/database/src/core/view/QueryParams.ts:182]: Were you planning to get an answer before checking in the code?

Comment (Feiyang1) [reply packages/database/src/core/view/QueryParams.ts:182]: Removed the comment. But what does it mean @schmidt-sebastian ?

File: packages/database/src/core/view/QueryParams.ts [hunk 2]
```diff
@@ -194,211 +198,237 @@ export class QueryParams {
...
-    }
+  return qs;
+}
 
-    if (this.endSet_) {
-      qs[REST_CONSTANTS.END_AT] = stringify(this.indexEndValue_);
-      if (this.endNameSet_) {
-        qs[REST_CONSTANTS.END_AT] += ',' + stringify(this.indexEndName_);
-      }
+export function queryParamsGetQueryObject(
```
Comment (schmidt-sebastian) [inline packages/database/src/core/view/QueryParams.ts:399]: FYI It looks like you re-ordered some of the methods, which makes this a internals impossible to review. I trust that you updated the methods correctly.

File: packages/database/src/core/view/QueryParams.ts (hunk 1 above)
Comment (schmidt-sebastian) [reply packages/database/src/core/view/QueryParams.ts]: It's a query with no filters and no constraints with the default ordering (by priority)

**Review by schmidt-sebastian — APPROVED** [review]

File: packages/database/src/core/view/QueryParams.ts (hunk 2 above)
Comment (Feiyang1) [reply packages/database/src/core/view/QueryParams.ts:399]: Tests are passing!

**Pull request merged**